src/components/JoinGroup/JoinGroupModal.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect } from 'react'
+import JoinGroupForm from './JoinGroupForm'
+import Modal from '../Modal/Modal'
+import { selectRegistrationToken, selectToken } from '../App/userSelector'
+import { useSelector, useDispatch } from 'react-redux'
+import { selectJoinGroupModal } from './joinGroupModalSelector'
+import { hideJoinGroupModal } from './joinGroupModalActions'
+import { Api } from '../../services/api'
+import {
+    showErrorMessage,
+    showSuccessMessage,
+} from '../App/Snackbar/snackbarActions'
+import { useTranslation } from 'react-i18next'
+
+const JoinGroupModal = () => {
+    const { isDisplayed } = useSelector(selectJoinGroupModal)
+    const dispatch = useDispatch()
+    const { t } = useTranslation('join-group-form')
+    const token = useSelector(selectToken)
+    const registrationToken = useSelector(selectRegistrationToken)
+
+    useEffect(() => {
+        if (!registrationToken && isDisplayed) {
+            dispatch(showErrorMessage(t('missing-registration-token')))
+        }
+    })
+
+    const onSubmit = async (values, { setSubmitting }) => {
+        const { name: groupName, password: groupPassword } = values
+        console.log('values:', values)
+        setSubmitting(true)
+        try {
```

Comment:
we usually have everything there is an action like `joinGroup` that manage all dispatch, api call and so. (so the 7 line here and below). I've alreayd said if for other PR, it may be a good time to start using it no?